source/d3d9/buffer_detection.cpp

```diff
@@ -121,26 +129,48 @@ void reshade::d3d9::buffer_detection::on_clear_depthstencil(UINT clear_flags)
 	com_ptr<IDirect3DSurface9> depthstencil;
 	_device->GetDepthStencilSurface(&depthstencil);
 
-	if (depthstencil != _depthstencil_original && depthstencil != _depthstencil_replacement)
-		return; // Can only avoid clear of the replacement surface
-
-	auto &counters = _counters_per_used_depth_surface[_depthstencil_original];
+	depthstencil = (depthstencil == _depthstencil_replacement) ? _depthstencil_original : depthstencil;
 
-	// Ignore clears when there was no meaningful workload
-	// Also triggers when '_preserve_depth_buffers' is false, since no clear stats are recorded then
-	if (counters.current_stats.vertices <= 4 || counters.current_stats.drawcalls == 0)
+	if (depthstencil == nullptr || _depthstencil_replacement == nullptr || depthstencil != depthstencil_clear_index.first)
 		return;
 
-	counters.clears.push_back(counters.current_stats);
+	// always bound the stats to the original depthstencil surface
+	auto &counters = _counters_per_used_depth_surface[depthstencil];
 
-	if (depthstencil_clear_index == counters.clears.size())
+	// Update stats with data from previous frame
+	if (counters.current_stats.drawcalls == 0 && _first_empty_stats)
```

Comment:
Do we need this? I know this was necessary in D3D11 in some cases, but I'm no aware of cases in D3D9 and I like to avoid code that is not explicitly needed.

Reply:
Yes, it i necessary in some games, for instance Oblivion.

Reply:
I mean, with a step by step debug, it goes into this condition.

Reply:
I just realized that this does not actually make sense in D3D9, since the replacement surfaces are always cleared at the end of the frame (in `buffer_detection::reset`), so the only time this can be triggered by the game is if it clears a surface at the start of the frame before drawing anything. And that is not a case we want to capture, so this actually hurts more than it helps.